python/paddle/v2/trainer.py

```diff
@@ -167,7 +173,8 @@ def train(self, reader, num_passes=1, event_handler=None, feeding=None):
                         batch_id=batch_id,
                         cost=cost,
                         evaluator=batch_evaluator))
-                self.__parameter_updater__.finishBatch(cost)
+                if batch_id >= step_size and batch_id % step_size == 0:
+                    self.__parameter_updater__.finishBatch(cost)
```

Comment:
This is assuming the gradient is not reset to 0 and always accumulating new gradients across different passes, unless `para->getBuf(PARAMETER_GRADIENT)->zeroMem();` is called.
Did you verify with @jacquesqiao or @hedaoyuan about the correctness of this implementation?